ecs-cli/modules/cli/local/project/project.go

```diff
@@ -0,0 +1,214 @@
+	}
+
+	p.taskDefinition = taskDefinition
+	return nil
+}
+
+func (p *localProject) readTaskDefinitionFromFile(filename string) (*ecs.TaskDefinition, error) {
+	bytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading task definition from %s: %s", filename, err.Error())
+	}
+
+	taskDefinition := ecs.TaskDefinition{}
+	err = json.Unmarshal(bytes, &taskDefinition)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing task definition JSON: %s", err.Error())
+	}
+
+	return &taskDefinition, nil
+}
+
+var newCommandConfig = func(context *cli.Context, rdwr config.ReadWriter) (*config.CommandConfig, error) {
+	return config.NewCommandConfig(context, rdwr)
+}
+
+// FIXME: NOTE this will actually read from either ARN or Task Definition family name
+func (p *localProject) readTaskDefinitionFromArn(arn string) (*ecs.TaskDefinition, error) {
+	rdwr, err := config.NewReadWriter()
+	if err != nil {
+		return nil, err
+	}
+
+	commandConfig, err := newCommandConfig(p.context, rdwr)
+	if err != nil {
+		return nil, err
+	}
+
+	ecsClient := ecsclient.NewECSClient(commandConfig)
+
+	return ecsClient.DescribeTaskDefinition(arn)
+}
+
+func (p *localProject) Convert() error {
+	// FIXME get secrets here, pass to converter?
+	data, err := converter.ConvertToDockerCompose(p.taskDefinition)
+
+	if err != nil {
+		return err
+	}
+
+	p.localBytes = data
+
+	return nil
+}
+
+func (p *localProject) Write() error {
+	// Will error if the file already exists, otherwise create
+
+	outputFileName := p.context.String(flags.LocalOutputFlag)
+	if outputFileName == "" {
+		outputFileName = LocalOutDefaultFileName
+	}
+	p.localOutFileName = outputFileName
+
+	out, err := os.OpenFile(outputFileName, os.O_WRONLY|os.O_CREATE|os.O_EXCL, LocalOutFileMode)
+	defer out.Close()
+
+	data := p.localBytes
+
+	if err != nil {
+		fmt.Printf("%s file already exists. Do you want to write over this file? [y/N]\n", outputFileName)
+
+		reader := bufio.NewReader(os.Stdin)
+		input, err := reader.ReadString('\n')
+		if err != nil {
+			return fmt.Errorf("Error reading input: %s", err.Error())
+		}
+
+		formattedInput := strings.ToLower(strings.TrimSpace(input))
+
+		if formattedInput != "yes" && formattedInput != "y" {
+			return fmt.Errorf("Aborted writing compose file. To retry, rename or move %s", outputFileName) // TODO add force flag
+		}
+
+		// Overwrite local compose file
+		err = ioutil.WriteFile(outputFileName, data, LocalOutFileMode)
+		return err
+	}
+
+	_, err = out.Write(data)
+
+	return err
+}
+
+// Get secret value stored in AWS Secrets Manager
+// TODO apply to each container
+func (p *localProject) getSecret(secretName string) (string, error) {
+	rdwr, err := config.NewReadWriter()
+	if err != nil {
+		return "", err
+	}
+
+	commandConfig, err := newCommandConfig(p.context, rdwr)
+	if err != nil {
+		return "", err
+	}
+
+	client := secretsmanager.NewSecretsManagerClient(commandConfig)
+
+	secret, err := client.GetSecretValue(secretName)
```

Comment:
dumb question: Is it okay for us to retrieve the value of their secret and output it in a file? Is there some sort of security concern here?

Reply:
The call is tied to their config, so nobody would be able to do this unless they have the customer's creds. No different from how you can display a secret value in the Secrets Manager console. But maybe ping @allisaurus on this?

Reply:
I share @efekarakus 's concerns re: outputting the secret contents to the file. Is there a way we can set the value as an env var and reference that var in the generated compose file?

Reply:
Writing to env vars and writing to a file aren't really that different from a security perspective --  you could argue writing to an environment variable where other processes might read from it might be worse. One thing that might mitigate this a bit is to change the file permissions to 400 or 600, so only the current user has read/write permissions and no one else does.

In any case, I don't believe there is significant risk, but we could add a log message that says that we'll be pulling and writing the decrypting a secret to the file just to make this more obvious.

Reply:
That makes sense! We'll consult with others offline to discuss what would be the right approach to move forward. 

One of the other recommendations suggested by @hencrice is to pass the secrets as environment variables to the `docker-compose up` command with `local up`. So something like `NAME=secret docker-compose -f docker-compose.local.yml up -d`. We'll check if this would be a good approach.

In the mean time to unblock you, what do you think about keeping the [client.GetSecretValue](https://github.com/aws/amazon-ecs-cli/pull/785/files/52b4b0d5a4161a9da22aa162b89736a24a1cd219?file-filters%5B%5D=.go&file-filters%5B%5D=.lock&file-filters%5B%5D=.yml&file-filters%5B%5D=No+extension#diff-11de9f88dfc38670546f5b1a4d5541faR101) as is because it's guaranteed to be used but deleting this function?